app/src/main/java/me/magnum/melonds/ui/cheats/CheatsFragment.kt

```diff
@@ -68,19 +70,19 @@ class CheatsFragment : Fragment() {
     }
 
     private fun openGameFoldersFragment() {
-        childFragmentManager.beginTransaction()
-                .setCustomAnimations(R.anim.fragment_translate_enter_push, R.anim.fragment_translate_exit_push, R.anim.fragment_translate_enter_pop, R.anim.fragment_translate_exit_pop)
-                .replace(R.id.layout_cheats_root, FoldersSubScreenFragment())
-                .addToBackStack(TAG_BACK_STACK_CHEATS)
-                .commit()
+        childFragmentManager.commit {
+            setCustomAnimations(R.anim.fragment_translate_enter_push, R.anim.fragment_translate_exit_push, R.anim.fragment_translate_enter_pop, R.anim.fragment_translate_exit_pop)
+            replace<FoldersSubScreenFragment>(R.id.layout_cheats_root)
+            addToBackStack(TAG_BACK_STACK_CHEATS)
+        }
     }
 
     private fun openCheatsFragment() {
-        childFragmentManager.beginTransaction()
-                .setCustomAnimations(R.anim.fragment_translate_enter_push, R.anim.fragment_translate_exit_push, R.anim.fragment_translate_enter_pop, R.anim.fragment_translate_exit_pop)
-                .replace(R.id.layout_cheats_root, CheatsSubScreenFragment())
-                .addToBackStack(TAG_BACK_STACK_CHEATS)
-                .commit()
+        childFragmentManager.commit {
+            setCustomAnimations(R.anim.fragment_translate_enter_push, R.anim.fragment_translate_exit_push, R.anim.fragment_translate_enter_pop, R.anim.fragment_translate_exit_pop)
+            replace<CheatsSubScreenFragment>(R.id.layout_cheats_root)
+            addToBackStack(TAG_BACK_STACK_CHEATS)
+        }
```

Comment:
I've now noticed that the code of these methods is duplicated. It would be nice to have a single generic method to avoid logic duplication. Something like this:

```kt
private inline fun <reified T : Fragment> openSubScreenFragment() {
    childFragmentManager.commit {
        setCustomAnimations(R.anim.fragment_translate_enter_push, R.anim.fragment_translate_exit_push, R.anim.fragment_translate_enter_pop, R.anim.fragment_translate_exit_pop)
        replace<T>(R.id.layout_cheats_root)
        addToBackStack(TAG_BACK_STACK_CHEATS)
    }
}
```

Then, the calls could be made as follows (from their respective methods):
```kt
openSubScreenFragment<FoldersSubScreenFragment>()

openSubScreenFragment<CheatsSubScreenFragment>()
```

Reply:
Noticed that too, but didn't know if I should have done anything about it lol.